parachain/pallets/incentivized-channel/src/inbound/mod.rs

```diff
@@ -106,3 +113,37 @@ decl_module! {
 		}
 	}
 }
+
+impl<T: Config> Module<T> {
+
+	fn handle_fee(amount: BalanceOf<T>, relayer: &T::AccountId) {
```

Comment:
@philipstanislaus plz review this function thoroughly too

Reply:
Just to explain this function a bit more (and I'll add a comment to the code as well):

* If the fee is zero, do nothing
* Otherwise, withdraw the fee amount from the Dot-App module account, returning a negative imbalance
* Figure out the fraction of the fee amount that should be paid to the relayer
* Pay the relayer if their account exists.
* Adjust the negative imbalance by offsetting the amount paid to the relayer
* Resolve the negative imbalance by depositing it into the treasury account

Here's a primer on imbalances to explain that aspect a bit more: https://blog.polymath.network/substrate-deep-dive-imbalances-8dfa89cc1d1

Reply:
This looks fine to me. Perhaps remove the `RewardRelayer` stuff from core though